src/herder/TransactionQueue.cpp

```diff
@@ -117,19 +242,19 @@ TransactionQueue::ban(std::vector<TransactionFramePtr> const& dropTxs)
 }
 
 bool
-TransactionQueue::contains(TransactionFramePtr tx)
+TransactionQueue::contains(TransactionFrameBasePtr tx)
 {
-    return find(tx).first != std::end(mPendingTransactions);
+    return find(tx).first != std::end(mAccountStates);
 }
 
 TransactionQueue::FindResult
-TransactionQueue::find(TransactionFramePtr const& tx)
+TransactionQueue::find(TransactionFrameBasePtr const& tx)
```

Comment:
I wonder if we should change this function to properly deal with fee bumped transactions, right now it's looking for an exact match but the use cases are:
* `TransactionQueue::extract` that wants to remove exact matches (so current behavior is desired)
* `TransactionQueue::contains` / `TransactionQueue::canAdd`
   * when called with a normal transaction, should also match inner transactions from FeeBump transactions from the queue, as the caller is just trying to answer the question that a given transaction is already or not in the queue
   * when called on a fee bumped transaction, we want exact match (current behavior)

Reply:
I think this is the last item left that will ensure that we don't interfere with clients that don't use `FeeBump`